src/FakeItEasy.Analyzer.Shared/NonVirtualSetupAnalyzer.cs

```diff
@@ -21,7 +21,13 @@
 #endif
     internal class NonVirtualSetupAnalyzer : DiagnosticAnalyzer
     {
-        private static readonly ImmutableHashSet<string> DiagnosticsMap = CreateDiagnosticsMap();
+        private static readonly ImmutableHashSet<string> CallSpecMethods =
+            ImmutableHashSet.Create(
+                StringComparer.Ordinal,
+                "FakeItEasy.A.CallTo",
+                "FakeItEasy.A.CallTo`1",
+                "FakeItEasy.A.CallToSet`1",
+                "FakeItEasy.Fake`1.CallsTo`1");

```

Comment:
Not related to the PR (I can move it to a different PR if you prefer), but the notion of "diagnostic map" made no sense in this case. It's a list, not a map, and we don't need a "diagnostic map", since this analyzer only produces one diagnostic.